package/libs/openssl/Makefile

```diff
@@ -100,18 +135,74 @@ $(call Package/openssl/Default/description)
 This package contains the OpenSSL command-line utility.
 endef
 
+define Package/libopenssl-gost
+  $(call Package/openssl/Default)
+  SUBMENU:=SSL
+  TITLE:=Russian GOST algorithms engine
+  DEPENDS:=libopenssl @OPENSSL_ENGINE @OPENSSL_USE_VERSION_1_0_2 +@OPENSSL_WITH_GOST
+endef
+
+define Package/libopenssl-gost/description
+This package adds an engine that enables Russian GOST algorithms.
+To use it, you need to configure it in /etc/ssl/openssl.cnf.
+See https://www.openssl.org/docs/man1.0.2/apps/config.html#ENGINE-CONFIGURATION-MODULE
+The engine_id is "gost"
+endef
+
+define Package/libopenssl-afalg
+  $(call Package/openssl/Default)
+  SUBMENU:=SSL
+  TITLE:=AFALG hardware acceleration engine
+  DEPENDS:=libopenssl @OPENSSL_ENGINE @!OPENSSL_USE_VERSION_1_0_2 +@KERNEL_AIO \
+	   @!LINUX_3_18 +kmod-crypto-user
```

Comment:
Please do not enable `KERNEL_AIO` at least not yet and not as part of the initial OpenSSL bump. Or alternatively restrict it to the more capable x86 and arm targets.

Reply:
I'll try to resolve the dependency upstream.    We can't even enable openssl aio support unless we use glibc, but the current engine requires kernel aio nonetheless.  So I think mandatory kernel aio is actually a bug, as aio support should not be required.  Even though it is a nice idea, it appears to slow things down when if you are not using async io (most of observed cases so far).  For the time being, I'll dump the selection.

Reply:
Thank you! :)